Soulifies Backpacks & Duffels

Sun-Soaked opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Backpacks and Duffel bags can now carry bulky items, and Duffel bags now have more storage space. 

Backpacks cannot be opened while worn on your back.
Duffels now cannot be opened unless placed on the ground. 
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
This gives satchel/messenger bag's low storage space an upside(access from back), while justifying putting bigger items in storage objects that really should be able to carry them.
Duffel bags are now very well-storage spaced, and can be specialized as a kind of stash bag. 

Backpack and duffel bags may need to have even more storage space to compensate, but for now I've been pretty frugal in buffing it. 
It's soul. Soon you will understand my vision
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
balance: Backpacks can now carry Bulky items, but can no longer be accessed while on your back.
balance: Duffel bags can now carry more, but must be placed on the ground to be opened.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sun-Soaked committed:
IT'S GOOD FOR YOU

take your medicine

SomeguyManperson commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -39,6 +39,7 @@
 	STR.max_volume = STORAGE_VOLUME_BACKPACK
 	STR.max_w_class = MAX_WEIGHT_CLASS_BACKPACK
 	STR.use_sound = 'sound/items/storage/unzip.ogg'
+	STR.worn_access = FALSE
```
satchels are a subtype of backpacks & need stuff updated

Sun-Soaked replied:
👍

Sun-Soaked committed:
assorted fixes

SomeguyManperson commented on code/game/objects/items/storage/backpack.dm:
```diff
@@ -186,6 +187,7 @@
 	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
 	STR.max_volume = STORAGE_VOLUME_BACKPACK
```
```suggestion
	STR.max_volume = STORAGE_VOLUME_SATCHEL
```

Sun-Soaked committed:
Update code/game/objects/items/storage/backpack.dm

Co-authored-by: Theos <theubernyan@gmail.com>
Signed-off-by: Sun-Soaked <45698967+Sun-Soaked@users.noreply.github.com>

Sun-Soaked committed:
satchel buff

technically everything buff but I needed to preserve -some- scaling

Sun-Soaked committed:
nerf + satchelifies messenger bags + examine

Sun-Soaked committed:
maybe

Sun-Soaked committed:
ughhh

thearbiber commented:
The lowered storage for satchels feels a bit worse to play with. The duffels and all that are pretty useful tools now.

Sun-Soaked commented:
> The lowered storage for satchels feels a bit worse to play with. The duffels and all that are pretty useful tools now.

satchel storage wasn't actually changed. check the changed files

thgvr commented:
placebo is real

thearbiber commented:
I may have gaslit myself

SomeguyManperson commented:
it was bc this actually swaps them to use satchel storage size from backpack storage size
placebo is FAKE

Sun-Soaked commented:
> it was bc this actually swaps them to use satchel storage size from backpack storage size placebo is FAKE

fuck

Ratvarr commented:
![image](https://github.com/user-attachments/assets/b4099aeb-0d36-4fc8-ac90-0de13a81fe65)
Need I say more
![image](https://github.com/user-attachments/assets/dedd389a-e74b-4029-aac1-6c8dcc0493b7)
Dufflebags.

Jedi-Toothpaste commented:
> ![image](https://private-user-images.githubusercontent.com/61395690/392645456-b4099aeb-0d36-4fc8-ac90-0de13a81fe65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMzNzA1NzQsIm5iZiI6MTczMzM3MDI3NCwicGF0aCI6Ii82MTM5NTY5MC8zOTI2NDU0NTYtYjQwOTlhZWItMGQzNi00ZmM4LWFjOTAtMGRlMTNhODFmZTY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDAzNDQzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxZTAyNWQ2NTQ0NjJhNGY3OWNjZGZmNjM3N2I1YWM2M2JiYjMxMjlkZTUwYjc1OTg2MjMzODBmNzMxNDNlYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CxTniKN-GiDvEEu5X_5rWSu0IgOYMrScuuPsOlA1Ijk) Need I say more ![image](https://private-user-images.githubusercontent.com/61395690/392645490-dedd389a-e74b-4029-aac1-6c8dcc0493b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMzNzA1NzQsIm5iZiI6MTczMzM3MDI3NCwicGF0aCI6Ii82MTM5NTY5MC8zOTI2NDU0OTAtZGVkZDM4OWEtZTc0Yi00MDI5LWFhYzEtNmM4ZGNjMDQ5M2I3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDAzNDQzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkZWY0N2YwZjVjYWM0NDk0MDE1ZjYyMGUxMmRiZWIwMDVjNmY5YmFmOWM4MzY2YTg3NTg5YjU0MWY5Y2M4NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IXSUnkB-jY2aci7D9SuCASTJzzp0RwE3KkC8eypm2ZA) Dufflebags.

I thought the storage would have been reduced since before duffelbags could only hold four bulky items, but I don't think the amount it could store was changed?

EDIT: ignore me i'm stupid, I think duffels should hold less actually

edit 2: I commented this on the wrong PR but you saw it anyway:
you can take your backpack off, hold the inventory open and re-equip it. Allowing you to take items out without taking it back off

Sun-Soaked commented:
> ![image](https://private-user-images.githubusercontent.com/61395690/392645456-b4099aeb-0d36-4fc8-ac90-0de13a81fe65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MTczOTgsIm5iZiI6MTczMzQxNzA5OCwicGF0aCI6Ii82MTM5NTY5MC8zOTI2NDU0NTYtYjQwOTlhZWItMGQzNi00ZmM4LWFjOTAtMGRlMTNhODFmZTY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDE2NDQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiNWE0N2NkOGFlMWVjYmUwM2FiOGQzN2EzZTdkZWUyZjI1OWVmODc2ODNjYWU0NmMzNGJhOTA3MmZhMTVjMDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.clcJ68n1_WWNB0WpqExOZ3z26V6ZbyMPYCO0nQr5zGk) Need I say more ![image](https://private-user-images.githubusercontent.com/61395690/392645490-dedd389a-e74b-4029-aac1-6c8dcc0493b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MTczOTgsIm5iZiI6MTczMzQxNzA5OCwicGF0aCI6Ii82MTM5NTY5MC8zOTI2NDU0OTAtZGVkZDM4OWEtZTc0Yi00MDI5LWFhYzEtNmM4ZGNjMDQ5M2I3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDE2NDQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZGE1MmQ3ZWEwNDk4YTE2N2VjMDA2YWZmMTg1MDBjMjQ3MTcwZDA0YjhjYjQxZGRmOWZjOGIyMWRmNGU3NDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZKnO8NG8GjAaMVv-vB_blrF2kKLce9rFqe_i3XtqJqE) Dufflebags.

top guns should not be tiny size- also duffels have been nerfed since this-
but this is also kind of inevitable as a consequence of Real item sizes

a storage object than can hold 5 bulky items can hold 10 normal ones, or 20 small ones, and so on. 
We'll need to do more sanity adjustment to pull a bunch of items up from tiny into small and certain objects from bulky into huge.

This only increased the storage of duffels by 1 bulky item, so this was already pretty much an issue

Jedi-Toothpaste commented:
followup to my issue from wednesday
You can still keep the inventory open and pull items out of a backpack/duffel
HOWEVER
you can put items back into inventories inside of the bag, for instance, you can put items into a box from the backpack
![image](https://github.com/user-attachments/assets/b0d1e5f0-d087-4a31-ad3e-9740bd3b91a6)
![image](https://github.com/user-attachments/assets/96f3c09e-315a-455e-9c17-06d4d484d73f)
![image](https://github.com/user-attachments/assets/1b1c6cb5-7c7f-4f2a-bd2b-26ea49940b7a)

and of course, you can take it out as the inventory is open already, and furthermore, because the box doesn't do the backpack check, you can shuffle items in and out allowing you to use it like a very crap storage

![image](https://github.com/user-attachments/assets/102bf428-6cb2-4323-8d66-4e363ebf4fda)

lastly, since the box storage is bigger it's actually viable to do this

generalthrax commented:
I have to say, satchel being 4 normal-sized items only feels. Very bad. Like terribly so.

thearbiber commented:
Satchels now hold 4 normal items. This is painful.

Sun-Soaked committed:
fixes backpack access

goongaga

Sun-Soaked commented:
if anyone figures out what this is or it appears in testing from a specific Thing ingame please let me know bc I can't reproduce it & it's apparently coming by way of like /obj/machinery/examine, and I have no idea how that's happening

![image](https://github.com/user-attachments/assets/db8cca33-c14a-4c9c-970f-96d85aa4618b)

Sun-Soaked committed:
refactors locked into check_access()

also fixes potential problem with showing messages to correct mob during storage procs

generalthrax commented:
Seems duffelbags can no longer be accessed even on the ground. Rations similarly can no longer be opened.

Sun-Soaked committed:
Revert "refactors locked into check_access()"

This reverts commit 62229db21d28ec1b9cd742452c6633cdd49bc348.

Sun-Soaked committed:
Merge branch 'master' into backpack_soul

Sun-Soaked commented:
> Seems duffelbags can no longer be accessed even on the ground. Rations similarly can no longer be opened.

issue with an access_check() refactor I did. should be fine now

tantspants commented:
Backpacks can't be accessed on the ground. It gives the same message as when you try when it's on your back.

Sun-Soaked committed:
toxic code yuri

bottom text

randy10122 commented:
hot take: i think the backpack changes are not good for gameplay at all
duffel changes are whatever - theyve always been the Niche Choice in bags
but why change backpacks? the way i see it, the only difference between a backpack and a satchel should be visual. why sacrifice style for changes that, ill be completely honest, i have only heard complaints about?
